src/V3RandomizeMethod.cpp

```diff
@@ -0,0 +1,234 @@
+            new AstConst(nodep->fileline(), AstConst::WidthedValue(), 32, 1)));
+        auto* classp = nodep;
+        for (auto* classp = nodep; classp;
+             classp = classp->extendsp() ? classp->extendsp()->classp() : nullptr) {
+            for (auto* memberp = classp->stmtsp(); memberp; memberp = memberp->nextp()) {
+                auto* memberVarp = VN_CAST(memberp, Var);
+                if (!memberVarp || !memberVarp->isRand()) continue;
+                AstNode* stmtp = nullptr;
+                if (VN_IS(memberp->dtypep()->skipRefp(), BasicDType)
+                    || VN_IS(memberp->dtypep()->skipRefp(), StructDType)) {
+                    auto* refp = new AstVarRef(nodep->fileline(), memberVarp, VAccess::WRITE);
+                    stmtp = newStdRandomize(nodep->fileline(), refp);
+                } else if (auto* classRefp = VN_CAST(memberp->dtypep(), ClassRefDType)) {
+                    auto* refp = new AstVarRef(nodep->fileline(), memberVarp, VAccess::WRITE);
+                    auto* funcp = V3RandomizeMethod::newRandomizeFunc(classRefp->classp());
+                    auto* callp = new AstMethodCall(nodep->fileline(), refp, "randomize", nullptr);
+                    callp->taskp(funcp);
+                    callp->dtypeFrom(funcp);
+                    stmtp = callp;
+                } else {
+                    memberp->v3warn(E_UNSUPPORTED,
+                                    "Unsupported: random member variables with type "
+                                        << memberp->dtypep()->skipRefp()->type().ascii());
+                }
+                if (stmtp) {
+                    // Although randomize returns int, we know it is 0/1 so can use faster
+                    // AstAnd vs AstLogAnd.
+                    stmtp = new AstAnd(nodep->fileline(),
+                                       new AstVarRef(nodep->fileline(), fvarp, VAccess::READ),
+                                       stmtp);
+                    auto* assignp = new AstAssign(
+                        nodep->fileline(), new AstVarRef(nodep->fileline(), fvarp, VAccess::WRITE),
+                        stmtp);
+                    funcp->addStmtsp(assignp);
+                }
+            }
+        }
+        nodep->user1(false);
+    }
+
+    virtual void visit(AstNode* nodep) override { iterateChildren(nodep); }
+
+public:
+    // CONSTRUCTORS
+    explicit RandomizeMethodVisitor(AstNetlist* nodep) { iterate(nodep); }
+    virtual ~RandomizeMethodVisitor() override = default;
+};
+
+//######################################################################
+// Randomize method class functions
+
+void V3RandomizeMethod::randomizeNetlist(AstNetlist* nodep) {
+    UINFO(2, __FUNCTION__ << ": " << endl);
+    {
+        RandomizeMethodMarkVisitor markVisitor(nodep);
+        RandomizeMethodVisitor visitor(nodep);
+    }
+    V3Global::dumpCheckGlobalTree("randomize_method", 0,
```

Comment:
```suggestion
    V3Global::dumpCheckGlobalTree("randomize", 0,
```

Please rename V3RandomizeMethod to V3Randomize, unless you anticipate needing another Randomize.

Reply:
Perhaps randomize constraints should be handled separately? (it also concerns `std::randomize`) Then the distinction between randomize method and randomize constraints would be clearer.

Reply:
Maybe.  Let's rename so it's shorter, we can always have another longer one also.